ui/elements.py

```
@@ -123,42 +126,48 @@ def __init__(self, controller, parent=None):
         layout.addWidget(self.start_stop)
         layout.addWidget(self.record)
         # update button based on digitiser state
+
         self.update()
     
 
-    def update(self):
+    def update(self): 
         '''
         Update the acquisition status based on the digitiser state.
         '''
-        if self.controller.digitiser is None:
+        try:
+            self.start_stop.clicked.disconnect()
+            self.record.clicked.disconnect()
+        except TypeError:
+            pass
```

Maybe add a comment here about how you're disconnecting the button's functionality before resetting them. This is a bit obtuse to those not familiar to PySide6

---

We want to safely disconnect previous button signal connections. If no connections exist, PyQt will throw a TypeError exception. By calling pass here, I am explicitly saying that this is fine since we update the button signals later in the function. Therefore, we can ignore the error. I have added a couple of comments in the code to explain this.